commands/utils/quick-pick-azure.ts

```diff
@@ -7,58 +7,60 @@ import { ResourceGroup } from 'azure-arm-resource/lib/resource/models';
+import { ResourceManagementClient } from '../../node_modules/azure-arm-resource';
 import * as acrTools from '../../utils/Azure/acrTools';
+import { isValidAzureName } from '../../utils/Azure/common';
 import { AzureImage } from "../../utils/Azure/models/image";
 import { Repository } from "../../utils/Azure/models/repository";
 import { AzureUtilityManager } from '../../utils/azureUtilityManager';
 
 export async function quickPickACRImage(repository: Repository, prompt?: string): Promise<AzureImage> {
-    const placeHolder = prompt ? prompt : 'Choose image to use';
+    const placeHolder = prompt ? prompt : 'Select image to use';
     const repoImages: AzureImage[] = await acrTools.getImagesByRepository(repository);
     const imageListNames = repoImages.map(img => <IAzureQuickPickItem<AzureImage>>{ label: img.tag, data: img });
     let desiredImage = await ext.ui.showQuickPick(imageListNames, { 'canPickMany': false, 'placeHolder': placeHolder });
     return desiredImage.data;
 }
 
 export async function quickPickACRRepository(registry: Registry, prompt?: string): Promise<Repository> {
-    const placeHolder = prompt ? prompt : 'Choose registry to use';
+    const placeHolder = prompt ? prompt : 'Select repository to use';
     const repositories: Repository[] = await acrTools.getRepositoriesByRegistry(registry);
     const quickPickRepoList = repositories.map(repo => <IAzureQuickPickItem<Repository>>{ label: repo.name, data: repo });
     let desiredRepo = await ext.ui.showQuickPick(quickPickRepoList, { 'canPickMany': false, 'placeHolder': placeHolder });
     return desiredRepo.data;
 }
 
 export async function quickPickACRRegistry(canCreateNew: boolean = false, prompt?: string): Promise<Registry> {
-    const placeHolder = prompt ? prompt : 'Choose registry to use';
+    const placeHolder = prompt ? prompt : 'Select registry to use';
     let registries = await AzureUtilityManager.getInstance().getRegistries();
-    interface RegistryWrapper {
-        isCreateNew?: boolean;
-        registry?: Registry;
-    }
-    let quickPickRegList: IAzureQuickPickItem<RegistryWrapper>[] = [];
-    if (canCreateNew) {
-        quickPickRegList.push(<IAzureQuickPickItem<RegistryWrapper>>{ label: "+ Create a new registry", data: { isCreateNew: true } });
-    }
-    quickPickRegList = quickPickRegList.concat(registries.map(reg => <IAzureQuickPickItem<RegistryWrapper>>{ label: reg.name, data: { registry: reg } }));
-    let desiredReg: IAzureQuickPickItem<RegistryWrapper> = await ext.ui.showQuickPick(quickPickRegList, {
+    let quickPickRegList = registries.map(reg => <IAzureQuickPickItem<Registry | undefined>>{ label: reg.name, data: reg });
+
+    let createNewItem: IAzureQuickPickItem<Registry | undefined> = { label: '+ Create new registry', data: undefined };
+    if (canCreateNew) { quickPickRegList.unshift(createNewItem); }
+
+    let desiredReg: IAzureQuickPickItem<Registry | undefined> = await ext.ui.showQuickPick(quickPickRegList, {
```

Comment:
ext.ui.showQuickPick will never return undefined (it will throw UserCancelledError instead)

Reply:
That's for the type of data within IAzureQuickPickItem however.  That's only so I can potentially create a new registry if desired, it should have not have an effect on error checking neither is it used towards that effect.